### PR TITLE
Robot rune

### PR DESCRIPTION
**Pull request by RuneThe**
First version of Robots, not checked against board/tile for correct methods.
Please make request for change if necessary.

### CODE REVIEW

File: src/main/java/inf112/skeleton/app/interfaces/IRobot.java [hunk 1]
```diff
@@ -1,19 +1,19 @@
...
+
+import inf112.skeleton.app.enums.Direction;
+
+// RT initial setup jan 29/30 - 2019 - cont feb 12
 public interface IRobot {
 
-    public void move();         // int = -1,1,2,3 (steps)
-    public void turn();         // int = 90, 270, 180 degrees
-    public void getDirection(); // get current direction of robot as int (0, 90, 280, 270)
-    public void giveCards();    // give chosen object(with card ID) of 5 cards to robot, to put in RobotMemory
```
Comment (petterdaae) [inline src/main/java/inf112/skeleton/app/interfaces/IRobot.java:8]: Should this method have a parameter. For example giveCards(List<IProgramCard> cards); or something like that?

**Review by petterdaae — CHANGES_REQUESTED** [review]
Is there any reason that Robot is an abstract class?

File: src/main/java/inf112/skeleton/app/interfaces/IRobot.java (hunk 1 above)
Comment (RuneThe) [reply src/main/java/inf112/skeleton/app/interfaces/IRobot.java:8]: No should not be abstract.
All the above from IRobot is removed, as functionallity of moving robot is in board, also giveCard, as we discussed that maybe cards to program robot is in deck-object in user.
Above routines should be changed in branch, are they not?

Comment (petterdaae) [reply src/main/java/inf112/skeleton/app/interfaces/IRobot.java:8]: Oh, lol, did'nt realize that :laughing:

Comment (petterdaae) [reply src/main/java/inf112/skeleton/app/interfaces/IRobot.java:8]: I approve if you remove abstract ;)

**Review by AndreasCanAtakan — APPROVED** [review]
I agree with Petter, this look very good (albeit the the abstract classes)

**Pull request merged**